dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/service/SoftwareService.java

```diff
@@ -0,0 +1,171 @@
+        artifactId + "&repos=" + config.getRepository();
+    val url = new URL(versionsUrl);
+    val response = MAPPER.readValue(url, MavenArtifactResults.class);
+    return response.results;
+  }
+
+  public String getLatestVersionUrl(SoftwareProperties config) {
+    return getVersionUrl("%5BRELEASE%5D", config.getArtifactId(), config);
+  }
+
+  public String getLatestICGCGetVersionUrl(String os, SoftwareProperties config) {
+    List<ArtifactFolder> result = getVersions("icgc-get", config);
+
+    return getICGCGetVersionUrl("0.2.10", os, config);
+  }
+
+  public String getVersionChecksumUrl(String version, SoftwareProperties config) {
+    return getVersionUrl(version, config.getArtifactId(), config) + ".md5";
+  }
+
+  public String getVersionSignatureUrl(String version, SoftwareProperties config) {
+    return getVersionUrl(version, config.getArtifactId(), config) + ".asc";
+  }
+
+  public String getICGCGetVersionUrl(String version, String os, SoftwareProperties config) {
+    String classifier;
+    if (os.equals("linux")) {
+      classifier = config.getLinuxClassifier();
+    } else {
+      classifier = config.getOsxClassifier();
+    }
+    return getVersionUrl(version, config.getIcgcArtifactId(), config) + classifier + ".zip";
+  }
+
+  public String getVersionUrl(String version, String artifactId, SoftwareProperties config) {
+    if (artifactId.equals("icgc-get")) {
+      return getRepositoryUrl(artifactId, config) + "/" + artifactId + "/" + version + "/" + artifactId + "_v" + version
+          + "_";
+    }
+    val classifier = config.getClassifier();
+    return getGroupUrl(config) + "/" + artifactId + "/" + version + "/" + artifactId + "-" + version + "-"
+        + classifier + ".tar.gz";
+  }
+
+  private String getGroupUrl(SoftwareProperties config) {
+    val groupId = config.getGroupId();
+    return getRepositoryUrl("icgc-storage-client", config) + "/" + groupId.replaceAll("\\.", "/");
+  }
+
+  private String getRepositoryUrl(String artifactId, SoftwareProperties config) {
+    String repository;
+    if (artifactId.equals("icgc-get")) {
+      repository = config.getIcgcRepository();
+    } else {
+      repository = config.getRepository();
+    }
+    return mavenRepositoryUrl + "/simple/" + repository;
+  }
+
+  @Data
+  public static class MavenArtifactResults {
+
+    List<MavenArtifactVersion> results;
+
+  }
+
+  @Data
+  public static class MavenArtifactVersion {
+
+    String version;
+    boolean integration;
+
+  }
+
+  @Data
+  public static class ArtifactFolderResults {
+
+    String repo;
+    String Path;
+    String created;
+    String createdBy;
+    String lastModified;
+    String modifiedBy;
+    String lastUpdated;
+    List<ArtifactFolder> children;
+    String uri;
+
+  }
+
+  @Data
+  public static class ArtifactFolder {
+
+    String uri;
+    boolean folder;
+
+  }
+
+  public int compareVersion(String[] arg0, String[] arg1) {
```

Comment:
As I alluded to today, I think creating a `Version` class is the preferred approach here:

http://stackoverflow.com/a/11024200/527333